main.cpp

```diff
@@ -1 +1,303 @@
+    
+    if(!current[1])
+    {
+        free(current[0]);
+        free(current);
+        *head = nullptr;
+        head[0] = nullptr;
+    }
+
+    else
+    {
+        while(current[1]!=nullptr)
+        {
+            previous = current;
+            current = (char**)current[1];
+        }
+        previous[1] = nullptr;
+        free(current[0]);
+        free(current);
+       
+    }
+}
+
+// Returns the number of items in the list.
+// int StringListSize(char** list);  
+int StringListSize (char** list)
+{
+    if(!list) return 0;
+
+    int counter(0);
+    
+    while(list!=nullptr)
+    {
+        counter++;
+        list = (char**)list[1];
+    }
+    return counter;
+}
+
+// Returns the index position of the first occurrence of str in the list.     
+// int StringListIndexOf(char** list, char* str);        
+int StringListIndexOf(char** list, const char* str)
+{
+    if(!list || !str[0]) return -2;
+    
+    int counter(0);
+    
+    while(list!=nullptr)
+    {
+        if(!strcmp(list[0], str)) return counter;
+        counter++;
+        list = (char**)list[1];
+    }
+
+    return -1;
+}
+
+// Removes all occurrences of str in the list.
+// void StringListRemove(char*** list, String str);
+void StringListRemove(char*** list, const char* str)    
+{
+    if(!*list || !str[0]) return;
+
+    char** current(*list), **previous(nullptr);
+  
+    while (current!=nullptr)
+    {
+        if(current == *list && !strcmp(current[0], str))
+        {
+            *list = (char**)current[1];
+            free(current[0]);
+            free(current);
+            if(!*list) return;
+            current = *list;
+            
+        }
+
+        else if(!strcmp(current[0], str))
+        {
+            previous[1] = current[1];
+            free(current[0]);
+            free(current);
+            current = (char**)previous[1];
+        }
+
+        else 
+        {  
+            previous = current;
+            current = (char **)current[1];
+        }
+        
+    }
+
+}
+
+// Removes duplicate entries from the list.
+// void StringListRemoveDuplicates(char** list);
+void StringListRemoveDuplicates(char** list)
+{
+    if(!list) return;
+    char** current(list), **previous(nullptr);
+
+    while(current!=nullptr)
+    {
+        char** next((char**)current[1]);
+
+        StringListRemove(&next, current[0]);
+        previous = current;
+        previous[1] = (char*) next;
+        current = next;
+    }
+}
+
+// Replaces every occurrence of the before, in each of the string lists's strings, with after.
+// void StringListReplaceInStrings(char** list, char* before, char* after);
+void StringListReplaceInStrings(char** list, const char* before, const char* after)
+{
+    if(!list || !before[0] || !after[0]) return;
+
+    char** current(list);
+
+    while (current!=nullptr)
+    {
+        if (!strcmp(current[0], before))
+        {
+            if (strlen(after) != strlen(current[0]))
```

Comment:
Decreased performance. The 'strlen' function was called multiple times inside the body of a loop.